src/components/common/AutoSearch.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import styled from 'styled-components';
+import OutsideClickHandler from 'react-outside-click-handler';
+import { isEmpty } from 'lodash/fp';
+
+import { fetchSearchedRepos } from 'actions/issues.actions';
+
+import bookIcon from 'assets/images/book.svg';
+
+class AutoSearch extends Component {
+  state = {
+    isOpen: false,
+    searchValue: ''
+  };
+
+  handleSearch = e => {
+    const value = e.target.value;
+    this.setState({ searchValue: value });
+    this.props.fetchSearchedRepos(value);
+  };
+
+  handleSelect = path => {
+    this.props.history.replace(`/${path}/issues`);
```

Comment:
You shouldn't use replace when navigating to another page unless you specifically want to delete the user history. In general this is a very bad practice because the user loses the possibility to go back to pages he's been at.

There is almost never a good reason to do this, this is used mostly as a hack if the last url posts something to the server twice if the user clicks back. This scenario is still wrong because it could have been solved better with more UX planning

Reply:
100%